### PR TITLE
Add platform view support to flutter_tizen

### PR DESCRIPTION
**Pull request by bbrto21**
* This is from the webview_flutter plugin.
* Mimic original flutter lib's directory structure.
* Keep syncing `platform_view` to latest in `flutter`

Signed-off-by: Boram Bae <boram21.bae@samsung.com>

### CODE REVIEW

**Comment by bbrto21** [comment]
How about this PR? I just moved this from `webview_flutter` to flutter_tizen package. 
If this is merged, `webview_flutter` will depend on it.

**Review by bwikbs — APPROVED** [review]
Wow! It's the exactly same what I wanted! :+1:

**Comment by bbrto21** [comment]
> Wow! It's the exactly same what I wanted! +1

Thanks to @swift-kim

**Comment by swift-kim** [comment]
Please provide something like https://github.com/flutter-tizen/plugins/pull/54#pullrequestreview-625891709 so that we can track the original source code and make updates later. I think now is the good chance to sync the latest source code from the framework but not just copy the old webview_flutter's code.

**Comment by bbrto21** [comment]
@swift-kim I added sources and update implementations to the latest.

File: packages/flutter_tizen/lib/src/rendering/platform_view.dart [hunk 1]
```diff
@@ -0,0 +1,187 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: public_member_api_docs
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
```
Comment (swift-kim) [inline packages/flutter_tizen/lib/src/rendering/platform_view.dart:10]: Unused import: `scheduler.dart`

File: packages/flutter_tizen/lib/src/rendering/platform_view.dart [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+import '../services/platform_views.dart';
+
+/// Source: [_PlatformViewState] flutter/packages/flutter/lib/src/rendering/platform_view.dart
```
Comment (swift-kim) [inline packages/flutter_tizen/lib/src/rendering/platform_view.dart:14]: We may use this standard format:
```suggestion
/// Source: [_PlatformViewState] in flutter/src/rendering/platform_view.dart
```
Notice that the path starts with `flutter/src` (without `lib` within it).

File: packages/flutter_tizen/lib/src/services/platform_views.dart [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+    }
+    _platformViewCreatedCallbacks.clear();
+    _state = _TizenViewState.disposed;
+    PlatformViewsServiceTizen._instance._focusCallbacks.remove(viewId);
+  }
+}
+
+/// Controls a Tizen view that is composed using a GL texture.
+///
+/// This controller is created from the [PlatformViewsServiceTizen.initTizenView] factory,
```
Comment (swift-kim) [inline packages/flutter_tizen/lib/src/services/platform_views.dart:197]: Did you mean

```suggestion
/// Controls a Tizen view that is rendered as a texture.
///
/// This controller is typically created with [PlatformViewsServiceTizen.initTizenView].
```

File: packages/flutter_tizen/lib/src/widgets/platform_view.dart [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+/// Source: [_AndroidPlatformView] in flutter/lib/src/widgets/platform_view.dart
+class _TizenPlatformView extends LeafRenderObjectWidget {
+  const _TizenPlatformView({
+    required this.controller,
+    required this.hitTestBehavior,
+    required this.gestureRecognizers,
+    this.clipBehavior = Clip.hardEdge,
+  })  : assert(controller != null),
+        assert(hitTestBehavior != null),
+        assert(gestureRecognizers != null);
```
Comment (swift-kim) [inline packages/flutter_tizen/lib/src/widgets/platform_view.dart:181]:
```suggestion
        assert(gestureRecognizers != null),
        assert(clipBehavior != null);
```

File: packages/flutter_tizen/lib/src/widgets/platform_view.dart [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    this.onPlatformViewCreated,
+    this.hitTestBehavior = PlatformViewHitTestBehavior.opaque,
+    this.layoutDirection,
+    this.gestureRecognizers,
+    this.creationParams,
+    this.creationParamsCodec,
+    this.clipBehavior = Clip.hardEdge,
+  })  : assert(viewType != null),
+        assert(hitTestBehavior != null),
+        assert(creationParams == null || creationParamsCodec != null);
```
Comment (swift-kim) [inline packages/flutter_tizen/lib/src/widgets/platform_view.dart:32]:
```suggestion
        assert(creationParams == null || creationParamsCodec != null),
        assert(clipBehavior != null);
```

File: packages/flutter_tizen/lib/widget.dart [hunk 6]
```diff
@@ -0,0 +1,7 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
```
Comment (swift-kim) [inline packages/flutter_tizen/lib/widget.dart:1]: You may rename this file to `widgets.dart`.

File: packages/flutter_tizen/lib/widget.dart [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+library widgets;
```
Comment (swift-kim) [inline packages/flutter_tizen/lib/widget.dart:5]: You don't need this line.

File: packages/flutter_tizen/CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,7 @@
+## 0.2.0
+
+* Add platform_view for texture view.
```
Comment (swift-kim) [inline packages/flutter_tizen/CHANGELOG.md:3]: The meaning is not clear.

Comment (swift-kim) [reply packages/flutter_tizen/CHANGELOG.md]: What does "Add platform view" mean?

**Review by swift-kim — APPROVED** [review]

**Pull request merged**